src/tests/compiler/ast/utils/extendedCommentParserTests.ts

```diff
@@ -0,0 +1,273 @@
+                pos: 5,
+                end: 11
+            }]);
+        });
+
+        it("should only get the first comment on a line", () => {
+            const text = `/*a*/ //b\n/*c*//*d*/ //e`;
+            doStatementedTests(text, [{
+                kind: ts.SyntaxKind.MultiLineCommentTrivia,
+                pos: 0,
+                end: 5
+            }, {
+                kind: ts.SyntaxKind.MultiLineCommentTrivia,
+                pos: 10,
+                end: 15
+            }]);
+        });
+
+        describe("js docs", () => {
+            it("should not get js docs when there is a node below", () => {
+                doStatementedTests("/** a */\nlet a;", [{
+                    kind: ts.SyntaxKind.VariableStatement,
+                    pos: 0,
+                    end: 15
+                }]);
+            });
+
+            it("should get js docs when there are no nodes", () => {
+                doStatementedTests("/** a */", [{
+                    kind: ts.SyntaxKind.MultiLineCommentTrivia,
+                    pos: 0,
+                    end: 8
+                }]);
+            });
+
+            it("should get js docs when there are no nodes below", () => {
+                doStatementedTests("let a;\n/** a */", [{
+                    kind: ts.SyntaxKind.VariableStatement,
+                    pos: 0,
+                    end: 6
+                }, {
+                    kind: ts.SyntaxKind.MultiLineCommentTrivia,
+                    pos: 7,
+                    end: 15
+                }]);
+            });
+        });
+    });
+
+    describe("member tests", () => {
+        function doTest(text: string, expectedNodes: { kind: SyntaxKind; pos: number; end: number; }[]) {
+            const sourceFile = createSourceFile(text);
+            const container = getContainer(sourceFile)!;
+            const result = ExtendedCommentParser.getOrParseChildren(container, sourceFile);
+
+            assertEqual(expectedNodes, result);
+
+            function getContainer(node: ts.Node): ContainerNodes | undefined {
+                if (ts.isClassDeclaration(node)
+                    || ts.isEnumDeclaration(node)
+                    || ts.isInterfaceDeclaration(node)
+                    || ts.isClassExpression(node)
+                    || ts.isTypeLiteralNode(node)
+                    || ts.isObjectLiteralExpression(node)
+                    || ts.isCaseClause(node)
+                    || ts.isDefaultClause(node))
+                {
+                    return node;
+                }
+
+                return ts.forEachChild(node, getContainer);
+            }
+        }
+
+        it("should not get comments on the same line", () => {
+            doTest("class c {\n/*a*/p;/*b*/ //c\n}", [{
+                kind: ts.SyntaxKind.PropertyDeclaration,
+                pos: 9,
+                end: 17
+            }]);
+        });
+
+        it("should ignore comments that are on the opening brace", () => {
```

Comment:
I think comments before the closing brace should be parsed. For example:

```ts
class C {
    prop: string;
/* test */ }
```

Even though it's on the same line as the close brace, it feels like a comment statement.